MapboxGeocoder/MBGeocoder.swift

```diff
@@ -269,8 +280,25 @@ open class Geocoder: NSObject {
                     completionHandler(data)
                 }
             } catch {
-                DispatchQueue.main.async {
-                    errorHandler(error as NSError)
+                // Handle single & single batch geocoding queries
+                do {
```

Comment:
Per chat with @1ec5 we may want to consider doing a refactor of this gnarly do/try/catch block.